m4/ax_lib_bzip2.m4

```diff
@@ -165,9 +165,11 @@ AC_DEFUN([AX_LIB_BZIP2],
 @%:@include <bzlib.h>
                     ]],
                     [[
-    const char *version;
+    const char * version;
     
     version = BZ2_bzlibVersion();
+    if (version == 0)
+      ; // fixes error with -Werror=unused-but-set-variable on g++
```

Comment:
It's part of the existing autoconf macros by Hartmut Holzgraefe. Perhaps it needs a function call to check linking? I don't know autotools well enough be sure.

Our bzip2 library is specific to osm2pgsql. I looked for an alternative one that is actively maintained by someone who understands autotools well so we wouldn't have to, but couldn't. I did find an alternative zlib one in the autoconf archive, but couldn't get it working.